teps/0113-produce-results-from-matrixed-pipelinetasks.md

```diff
@@ -0,0 +1,204 @@
+and is already supported. However, producing `Results` from `PipelineTasks` with a `Matrix` was out of scope to await
+[TEP-0075: Object Parameters and Results][tep-0075] and [TEP-0076: Array Results][tep-0076]. This TEP aims to enable
+producing `Results` from `PipelineTasks` with a `Matrix` so that they can be used in subsequent `PipelineTasks`.
+
+### Use Cases
+
+#### 1. Building Images
+
+In [TEP-0090: Matrix][tep-0090], we described use cases for `Matrix` that involve building images - [kaniko][kaniko]
+and [monorepos][monorepos]. When the fanned out `PipelineTasks` produce the images as `Results`, the users would 
+need to pass them to subsequent `PipelineTasks` that to scan and deploy the images, among other operations. 
+
+To be specific, the [kaniko][kaniko] use case uses the [*kaniko*][kaniko-task] `Task` from the *Tekton Catalog*
+which produces an IMAGE-DIGEST `Result`. When the `PipelineTask` has a `Matrix`, it will be fanned out to multiple 
+`TaskRuns` to execute that `Task` - each of which will produce an IMAGE-DIGEST `Result`.
+
+#### 2. Testing
+
+In [TEP-0090: Matrix][tep-0090], we described use cases for `Matrix` that involve testing - [strategies][strategies]
+and [sharding][sharding]. When the fanned out `PipelineTasks` produce test outputs as `Results`, the users would 
+want to pass them to subsequent `PipelineTasks` that to process the test outputs.
+
+## Background
+
+This section describes the using `Results` in `Pipelines` - see [docs][results-docs] for further details.
+
+### String Results
+
+String `Results` are is a beta feature and is referred to as `$(tasks.<pipelinetask-name>.results.<result-name>)`.
+
+String `Results` from previous `TaskRuns` can be passed into the `Matrix` can be passed individually into the `Matrix`:
+
+```yaml
+tasks:
+...
+- name: task-4
+  taskRef:
+    name: task-4
+  matrix:
+  - name: values
+    value: 
+    - (tasks.task-1.results.foo) # string
+    - (tasks.task-2.results.bar) # string
+    - (tasks.task-3.results.rad) # string
+```
+
+### Array Results
+
+Array `Results` is an alpha feature and is referred to as `$(tasks.<pipelinetask-name>.results.<result-name>[*])`.
+Array indexing is supported and is referred as `$(tasks.<pipelinetask-name>.results.<result-name>[i])` where `i`
+is the index.
+
+Array `Results` from previous `TaskRuns` can be passed into the `Matrix` can be passed individually into the `Matrix`:
+
+```yaml
+tasks:
+...
+- name: task-5
+  taskRef:
+    name: task-5
+  matrix:
+  - name: values
+    value: (tasks.task-4.results.foo[*]) # array
+```
+
+For further information, see [TEP-0076: Array Results][tep-0076]. 
+
+### Object Results
+
+Object `Results` is an as alpha feature and is referred to as `$(tasks.<pipelinetask-name>.results.<result-name>[*])`. 
+Object elements can be referred to as `$(tasks.<pipelinetask-name>.results.<result-name>.key)`.
```

Comment:
Can object results be used in a matrix somehow?

Reply:
clarified this in the TEP